core/model.py

```diff
@@ -120,7 +121,7 @@ def is_saved(self) -> bool:
         '''
         Retrun True if this record is saved in database.
         '''
-        return hasattr(self, 'id') and self.id
+        return bool(hasattr(self, 'id') and self.id)
```

Comment:
I think `bool` is redundant here? `and` should always return a boolean

Reply:
Yep. My first thought is the same as yours, until I debug and found that it return an actual id as string if the record is saved.

Reply:
Ah, yes, indeed, the thing where you can use `or` to return the truthy variable. `and` seems to have the same thing.